packages/sanity/src/desk/structureBuilder/StructureNodes.ts

```diff
@@ -38,41 +41,57 @@ export interface EditorNode extends StructureNode {
   }
 }
 
-export interface Divider {
+/** @beta */
+export interface StructureDivider {
   id: string
   type: 'divider'
 }
 
-export type SerializePath = (string | number)[]
+/** @beta */
+export type StructureSerializePath = (string | number)[]
 
-export interface SerializeOptions {
-  path: SerializePath
+/** @beta */
+export interface StructureSerializeOptions {
+  path: StructureSerializePath
   index?: number
   hint?: string
 }
 
-export interface Serializable<T> {
-  serialize(options: SerializeOptions): T
+/** @beta */
+export interface StructureSerializable<T> {
+  serialize(options: StructureSerializeOptions): T
 }
 
-export type Collection = List | DocumentList | EditorNode | DocumentNode | Component
+/** @beta */
+export type StructureCollection =
+  | StructureList
+  | StructureDocumentList
+  | StructureEditorNode
+  | StructureDocumentNode
+  | StructureComponent
 
-export type CollectionBuilder =
-  | ListBuilder
-  | DocumentListBuilder
-  | DocumentTypeListBuilder
-  | DocumentBuilder
-  | ComponentBuilder
+/** @beta */
+export type StructureCollectionBuilder =
+  | StructureListBuilder
+  | StructureDocumentListBuilder
+  | StructureDocumentTypeListBuilder
+  | StructureDocumentBuilder
+  | StructureComponentBuilder
 
-export type Child = Collection | CollectionBuilder | ChildResolver
+/** @beta */
+export type StructureChild =
+  | StructureCollection
+  | StructureCollectionBuilder
+  | StructureChildResolver
 
-export type Builder =
-  | CollectionBuilder
-  | ComponentBuilder
-  | DocumentBuilder
-  | DocumentListBuilder
-  | DocumentListItemBuilder
-  | ListItemBuilder
-  | MenuItemBuilder
-  | MenuItemGroupBuilder
-  | InitialValueTemplateItemBuilder
+/** @internal */
+export type Structure__Builder =
```

Comment:
I have to admit, I find these changes kinda brutal. Kind of like a law passed for american congress where they add massive, unrelated bit on the end to get it through ;)

Kidding aside, any chance they could live in a separate PR? Or at least give a rundown of why this is necessary?

Reply:
Think about it – what's the difference between `Builder` and `StructureBuilder`? I didn't want to spend time figuring that out.